scripts/xe-toolstack-restart

```diff
@@ -32,6 +32,16 @@ SERVICES="perfmon xapi v6d xenopsd xenopsd-xc xenopsd-xenlight
   xcp-rrdd-xenpm xcp-rrdd-gpumon xcp-rrdd xcp-networkd squeezed forkexecd
   $MPATHALERT xapi-storage-script xapi-clusterd varstored-guard message-switch"
 
+tmp_file=$(mktemp)
+/opt/xensource/libexec/xapissl stop > $tmp_file 2>&1
+kill_stunnel_exit_code=$?
+if [[ $kill_stunnel_exit_code != 0 ]]; then
+  echo $'\nFailed to kill stunnel processes! Output:'
```

Comment:
What is `$` doing here? I have not seen used it like this.

Reply:
without the `$`, echo prints the `\n` literally as `\n` rather than a new line

Reply:
I would suggest to avoid this; instead:
```
echo
echo 'Failed to kill stunnel processes! Output:'
```

Reply:
COnsider `printf '\nFailed to kill stunnel processes! Output:'`

Reply:
i went with `printf`